src/google/appengine/ext/ndb/model.py

```diff
@@ -1913,7 +1913,10 @@ def _from_base_type(self, value):
       return pickle.loads(value)
     except UnicodeDecodeError:
       if int(os.environ.get('NDB_PY2_UNPICKLE_COMPAT', '0')):
-        return pickle.loads(value, encoding='bytes')
+        try:
+          return pickle.loads(value, encoding='bytes')
+        except:
+          return pickle.loads(value, encoding='latin1')
       raise
```

Comment:
Is encoding='bytes' ever what you would want here anyway? I would maybe try 'latin1' first, it's still going to mess up unicode strings but it's probably better for the common case of datetime objects.